src/app/modules/courses/+state/effects/courses-api-effects.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@angular/core';
+import { CoursesService } from '@modules/courses/services/courses.service';
+import { Actions, concatLatestFrom, createEffect, ofType } from '@ngrx/effects';
+import { CoursesActions, CoursesApiActions } from '@modules/courses/+state/actions';
+import { concatMap, exhaustMap, forkJoin, map, mergeMap, of, switchMap } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { selectAllCourses } from '@modules/courses/+state/reducers';
+
+@Injectable()
+export class CoursesApiEffectsService {
+
+  constructor(
+    private coursesService: CoursesService,
+    private actions$: Actions,
+    private store: Store
+  ) {
+  }
+
+  loadCourses$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(CoursesActions.enterToCoursesPage),
+      exhaustMap(() => {
+        return this.coursesService.fetchCourses().pipe(
+          map(courses => CoursesApiActions.coursesLoaded({ courses }))
+        );
+      })
+    );
+  });
+
+  loadMoreCourses$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(CoursesActions.loadMoreCourses),
+      concatLatestFrom(() => this.store.select(selectAllCourses)),
+      concatMap(([action, courses]) => {
```

Comment:
Looks like you are randomly use all types of Map here: exhaustMap, concatMap, mergeMap, switchMap. Looks strange